sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,497 @@
+
+                    # target number in filename that matches wavelength_center
+                    start = name.find('A')
+                    end = name.find('_F')
+                    wavelength = name[start + 1:end]
+
+                    # UV files have 'thick' files that need to be filtered out for no duplicates
+                    if int(wavelength) in self.wavelength_centers:
+
+                        # match the properties in the rec.array to desired properties
+                        variables = set(data[name][0].dtype.fields.keys()) & set(self.properties)
+
+                        # then iterate through the properties to fill a dataframe with rec.array information
+                        for property in variables:
+                            try:
+                                df.loc[property, wavelength] = data[name][0][property][0]
+                            # contam and crossarea are sequences -- # TODO: work on this try statement
+                            except ValueError:
+                                df.loc[property, wavelength] = data[name][0][property]
+                    else:
+                        pass
+
+        assert len(df) != 0, 'Data Frame is not loading from file.'
+
+        #  save to .csv outfile
+        df.to_csv('channel_properties.csv')
+
+        return df
+
+    def property_per_channel_dict(self, wavelength_center=94):
+        """
+        Returns a dictionary for the channel whose wavelength center is inputed.
+
+        Parameters
+        ----------
+        wavelength_center: int
+            has to be one of the wavelengths in self.wavelength_centers
+
+
+        Returns
+        -------
+        :returns A dictionary with the following keys and values:
+
+        wave: array
+            wavelengths imaged by the instrument and centered around the channel wavelength
+        geoarea: float
+            the geometric area of the each channel: EUV = 83.0 cm**2, UV = 30.8 cm**2
+        ent_filter: array
+            transmission efficiency of the entrance filter
+        fp_filter: array of floats
+            transmission efficiency of the focal plane filter
+        contam: array of floats
+            correction to adjust for the time varying contamination of the components
+        secondary: array of floats
+            reflectance of secondary mirror
+        primary: array
+            reflectance of the primary mirror
+        ccd: array, floats
+            the quantum efficiency of the ccd
+        """
+        assert wavelength_center in self.wavelength_centers, 'Choose one integer value from wavelength centers: ' + str(
+            self.wavelength_centers)
+
+        dict = {}
+
+        # target wavelength specifid and pull out respective properties in dataframe and apply units
+        for wavelength, values in self.dataframe.iteritems():
+            if int(wavelength) == int(wavelength_center):
+                dict['wave'] = values['wave'] * u.angstrom
```

Comment:
Instead of using the same keywords returned by the genx file, we should use more descriptive ones. For example, `geoarea` to `geometrical_area`, `wave` to `wavelength`, `fp_filter` to `focal_plane_filter`, etc. 

One of the weaknesses of these data structures returned by IDL is that it's confusing as to what data is stored where. We want to avoid these types of ambiguities.